_example/install.bat

```diff
@@ -1 +1,50 @@
 rem TODO
+@echo off
+rem get OS version
+for /f "tokens=4-5 delims=. " %%i in ('ver') do set VERSION=%%i.%%j
+if "%version%" == "6.3" set currentOS=win8.1
+if "%version%" == "6.2" set currentOS=win8
+if "%version%" == "6.1" set currentOS=win7
+if "%version%" == "6.0" set currentOS=winVista
+if "%version%" == "10.0" set currentOS=win10
+
+rem get this Computer Architecture
+set currentArch=%processor_architecture%
```

Comment:
Unfortunately this isn't always this simple. If you're running in a WOW64 process (x86 on x64, x86 on ARM64, ARM32 on ARM64) then there's another check needed here for `PROCESSOR_ARCHITEW6432`:
https://docs.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details

(Note that, for various reasons, PowerShell on ARM64 is x86.)